middleware/tests/admin/test_sgx_authorization.py

```diff
@@ -0,0 +1,333 @@
+
+        with patch("builtins.open", mock_open()) as open_mock:
+            open_mock.return_value.read.return_value = jsonsample
+
+            with self.assertRaises(ValueError):
+                SGXMigrationAuthorization.from_jsonfile("/an/existing/file.json")
+
+    def test_authorization_message(self):
+        # Test authorization message format
+        expected_msg = (b"\x19Ethereum Signed Message:\n" +
+                        b"160" +
+                        b"RSK_powHSM_SGX_upgrade_from_" +
+                        self.exporter_mrenclave.encode("ASCII") +
+                        b"_to_" + self.importer_mrenclave.encode("ASCII"))
+        self.assertEqual(expected_msg, self.sa.migration_spec.get_authorization_msg())
+
+    def test_authorization_digest(self):
+        # Test authorization digest generation
+        msg = self.sa.migration_spec.get_authorization_msg()
+        expected_digest = keccak_256(msg)
+        self.assertEqual(
+            expected_digest,
+            self.sa.migration_spec.get_authorization_digest()
+        )
+
+
+class TestMigrationSpec(TestCase):
+    def setUp(self):
+        # Sample mrenclave values (32-byte hex strings)
+        self.exporter_mrenclave = "aa" * 32
+        self.importer_mrenclave = "bb" * 32
+        self.migration_spec = SGXMigrationSpec({
+            "exporter": self.exporter_mrenclave,
+            "importer": self.importer_mrenclave
+        })
+
+    def test_mrenclave_getters(self):
+        # Test mrenclave getters and normalization
+        self.assertEqual(self.migration_spec.exporter, self.exporter_mrenclave.lower())
+        self.assertEqual(self.migration_spec.importer, self.importer_mrenclave.lower())
+
+    def test_invalid_exporter_mrenclave(self):
+        # Test constructor with invalid exporter mrenclave
+        with self.assertRaises(ValueError):
+            SGXMigrationSpec({
+                "exporter": "not-a-mrenclave",
+                "importer": self.importer_mrenclave
+            })
+
+    def test_invalid_importer_mrenclave(self):
+        # Test constructor with invalid importer mrenclave
+        with self.assertRaises(ValueError):
+            SGXMigrationSpec({
+                "exporter": self.exporter_mrenclave,
+                "importer": "not-a-mrenclave"
+            })
+
+    def test_mrenclave_normalization(self):
+        # Test mrenclave hex string normalization
+        spec = SGXMigrationSpec({
+            "exporter": "0x" + "aa" * 32,
+            "importer": "0x" + "bb" * 32
+        })
+        self.assertEqual(spec.exporter, "aa" * 32)
+        self.assertEqual(spec.importer, "bb" * 32)
+
+    def test_to_dict(self):
+        # Test dictionary conversion
+        expected_dict = {
+            "exporter": self.exporter_mrenclave,
+            "importer": self.importer_mrenclave,
+        }
+        self.assertEqual(expected_dict, self.migration_spec.to_dict())
+
+    def test_authorization_message(self):
```

Comment:
Should we also add an additional non-ethereum prefixed message generation test case?

Reply:
Sounds good, I'll add a test case for it.

Reply:
This will be tackled in an upcoming PR